ui/widgets/autocomplete.js

```diff
@@ -89,6 +90,20 @@ $.widget( "ui.autocomplete", {
 
 		this._addClass( "ui-autocomplete-input" );
 		this.element.attr( "autocomplete", "off" );
+		if ( this.options.textDir ) {
+			this.element.css( "text-align",
+				this.element.css( "direction" ) === "rtl" ? "right" : "left" );
+
+			this.element.css( "direction", this._getTextDir( this.element.text() ) );
+			if ( this.options.textDir === "auto" ) {
+				var that = this;
+				this.element.bind( "keyup", function() {
+					var fld = $( this );
+					fld.css( "direction",
+						that._getTextDir( that.valueMethod.apply( fld ) ) );
```

Comment:
I filter now the majority of irrelevant key strokes that have no imact on field content